R/module_nested_tabs.R

```diff
@@ -57,33 +57,45 @@ ui_teal_module.default <- function(id, modules, depth = 0L) {
 #' @export
 ui_teal_module.teal_modules <- function(id, modules, depth = 0L) {
   ns <- NS(id)
-  do.call(
-    tabsetPanel,
-    c(
-      # by giving an id, we can reactively respond to tab changes
-      list(
-        id = ns("active_tab"),
-        type = if (modules$label == "root") "pills" else "tabs"
-      ),
-      lapply(
-        names(modules$children),
-        function(module_id) {
-          module_label <- modules$children[[module_id]]$label
-          tabPanel(
-            title = module_label,
-            value = module_id, # when clicked this tab value changes input$<tabset panel id>
-            ui_teal_module(
-              id = ns(module_id),
-              modules = modules$children[[module_id]],
-              depth = depth + 1L
+  tags$div(
+    class = "teal-body",
```

Comment:
Maybe this `div(class = "teal-body, ...)` should be in `ui_teal`. Otherwise this div will be added as many times as depth-levels. `modules(label = "tab1", modules(label = "tab1.1, modules(tab = "1.1.1", ...`

Reply:
True, good catch. Let me move it.